rpcs3/Emu/Cell/SPURecompiler.cpp

```diff
@@ -6463,6 +6463,30 @@ class spu_llvm_recompiler : public spu_recompiler_base, public cpu_translator
 
 	void FSM(spu_opcode_t op)
 	{
+		// FSM following a comparison instruction
+		if (match_vr<s8[16], s16[8], s32[4], s64[2]>(op.ra, [&](auto c, auto MP)
+		{
+			using VT = typename decltype(MP)::type;
+
+			if (auto [ok, x] = match_expr(c, sext<VT>(match<bool[std::extent_v<VT>]>())); ok)
+			{
+				if constexpr (std::extent_v<VT> == 4) // u32[4]
+				{
+					set_vr(op.rt, (splat_scalar(c)));
```

Comment:
I think you can now use `splat_scalar(c)` regardless of VT.

Reply:
Ok, I just hope that the s16[8] case still generates as good code as for s8[16], I might check what it emits later.